### PR TITLE
Add support for logging of H2O MOJO Models

### PR DESCRIPTION
**Pull request by thebrianbn**
## What
  * Adds support for saving and logging of H2O MOJO models, in addition to the support of binary models
  * Update `get_data` to handle deserialization of H2O MOJO models
  * Unit tests updated/added to test for logging of H2O MOJO models

## How to Test
  * `python -m pytest tests/unit`
  * `python -m pytest tests/integration/test_schema.py`

### CODE REVIEW

**Review by dianelee217 — APPROVED** [review]

File: rubicon_ml/client/artifact.py [hunk 1]
```diff
@@ -119,12 +120,18 @@ def get_data(
             except Exception as err:
                 return_err = err
             else:
-                if deserialize == "h2o":
+                if deserialize == "h2o_binary":
```
[the remark this answers is not in the document]
Comment (thebrianbn) [reply rubicon_ml/client/artifact.py:123]: Yeah I was just thinking about this recently too. I will add that backwards compatibility!

File: rubicon_ml/client/artifact.py [hunk 2]
```diff
@@ -71,7 +71,7 @@ def _get_data(self):
     @failsafe
     def get_data(
         self,
-        deserialize: Optional[Literal["h2o", "pickle", "xgboost"]] = None,
+        deserialize: Optional[Literal["h2o_binary", "h2o_mojo", "pickle", "xgboost"]] = None,
```
Comment (ryanSoley) [inline rubicon_ml/client/artifact.py:74]:
```suggestion
        deserialize: Optional[Literal["h2o", "h2o_binary", "h2o_mojo", "pickle", "xgboost"]] = None,
```

File: tests/unit/client/test_artifact_client.py [hunk 3]
```diff
@@ -159,8 +160,18 @@ def test_download_location(mock_open, project_client):
     mock_file().write.assert_called_once_with(data)
 
 
+@pytest.mark.parametrize(
+    ["use_mojo", "deserialization_method"],
+    [
+        (False, "h2o_binary"),
+        (True, "h2o_mojo"),
```
Comment (ryanSoley) [inline tests/unit/client/test_artifact_client.py:167]:
```suggestion
        (True, "h2o"),
        (True, "h2o_mojo"),
```

File: rubicon_ml/client/artifact.py [hunk 4]
```diff
@@ -101,6 +102,13 @@ def get_data(
             )
             deserialize = "pickle"
 
+        if deserialize == "h2o":
+            warnings.warn(
+                "'deserialize' method 'h2o' will be deprecated in a future release,"
+                " please use 'h2o_binary' instead.",
+                DeprecationWarning,
```
Comment (ryanSoley) [inline rubicon_ml/client/artifact.py:109]: thanks!

**Review by ryanSoley — APPROVED** [review]
looks great!

**Pull request merged**